cpp/src/io/json/json_tree.cu

```diff
@@ -58,6 +58,50 @@
 
 namespace cudf::io::json {
 namespace detail {
+namespace {
+
+/**
+ * @brief Utility for calling thrust::copy_if
+ *
+ * Workaround for thrust::copy_if bug (https://github.com/NVIDIA/thrust/issues/1302)
+ * where it cannot iterate over int-max values `distance(first,last) > int-max`
+ * This calls thrust::copy_if in 2B chunks instead.
+ */
+template <typename InputIterator,
+          typename StencilIterator,
+          typename OutputIterator,
+          typename Predicate>
+OutputIterator thrust_copy_if(rmm::exec_policy policy,
+                              InputIterator first,
+                              InputIterator last,
+                              StencilIterator stencil,
+                              OutputIterator result,
+                              Predicate pred)
+{
+  auto const copy_size = std::min(static_cast<std::size_t>(std::distance(first, last)),
+                                  static_cast<std::size_t>(std::numeric_limits<int>::max()));
+
+  auto itr = first;
+  while (itr != last) {
+    auto const copy_end =
+      static_cast<std::size_t>(std::distance(itr, last)) <= copy_size ? last : itr + copy_size;
+    result = thrust::copy_if(policy, itr, copy_end, stencil, result, pred);
+    stencil += std::distance(itr, copy_end);
+    itr = copy_end;
+  }
+  return result;
+}
+
+template <typename InputIterator, typename OutputIterator, typename Predicate>
+OutputIterator thrust_copy_if(rmm::exec_policy policy,
+                              InputIterator first,
+                              InputIterator last,
+                              OutputIterator result,
+                              Predicate pred)
+{
+  return thrust_copy_if(policy, first, last, first, result, pred);
```

Comment:
Wow, I assume that you're going to extract this out into a common utility file and call it across many places.

Reply:
https://github.com/rapidsai/cudf/pull/12190#issuecomment-1332799678